applications/mne_x/plugins/noisereduction/noisereduction.cpp

```diff
@@ -295,127 +306,74 @@ void NoiseReduction::showOptionsWidget()
-        write_eigen_matrix(matSpharaMultMag, QString(QCoreApplication::applicationDirPath() + "/mne_x_plugins/resources/noisereduction/SPHARA/matSpharaMultMag.txt"));
-        write_eigen_matrix(m_matSpharaMultVVMag, QString(QCoreApplication::applicationDirPath() + "/mne_x_plugins/resources/noisereduction/SPHARA/m_matSpharaMultVVMag.txt"));
+        m_matSpharaMultFirst = Sphara::makeSpharaProjector(m_matSpharaVVGradLoaded, indicesFirstVV, m_pFiffInfo->nchan, m_iNBaseFctsFirst, 1); //GRADIOMETERS
+        m_matSpharaMultSecond = Sphara::makeSpharaProjector(m_matSpharaVVMagLoaded, indicesSecondVV, m_pFiffInfo->nchan, m_iNBaseFctsSecond, 0); //Magnetometers
     }
 
     if(m_sCurrentSystem == "BabyMEG") {
-        //INNER LAYER
-        //Remove unwanted base functions
-        qDebug()<<"NoiseReduction::creatSpharaOperator - Removing"<<m_matSpharaMultBMInner.rows() - m_iNBaseFcts<<"SPHARA grad base functions";
-
-        MatrixXd matSpharaInnerCut = m_matSpharaBabyMEGInnerFull.block(0,0,m_matSpharaBabyMEGInnerFull.rows(),m_iNBaseFcts);
-        MatrixXd matSpharaMultInner = matSpharaInnerCut * matSpharaInnerCut.transpose().eval();
-
-        //Create full gradiometer SPHARA operator
-        int rowIndex = 0;
-        int colIndex = 0;
-        for(int r = 0; r<m_matSpharaMultBMInner.rows(); r++) {
-            //Scan for inner layer channels
-            if(m_pFiffInfo->chs.at(r).coil_type == 7002) {
-                for(int c = 0; c<m_matSpharaMultBMInner.cols(); c++) {
-                    if(m_pFiffInfo->chs.at(c).coil_type == 7002) {
-                        m_matSpharaMultBMInner(r,c) = matSpharaMultInner(rowIndex,colIndex);
-                        colIndex++;
-                    }
-                }
-                colIndex = 0;
-                rowIndex++;
-            }
-        }
-
-        //Write final operator matrices to file
-        write_eigen_matrix(matSpharaInnerCut, QString(QCoreApplication::applicationDirPath() + "/mne_x_plugins/resources/noisereduction/SPHARA/matSpharaInnerCut.txt"));
-        write_eigen_matrix(matSpharaMultInner, QString(QCoreApplication::applicationDirPath() + "/mne_x_plugins/resources/noisereduction/SPHARA/matSpharaMultInner.txt"));
-        write_eigen_matrix(m_matSpharaMultBMInner, QString(QCoreApplication::applicationDirPath() + "/mne_x_plugins/resources/noisereduction/SPHARA/m_matSpharaMultBMInner.txt"));
-        write_eigen_matrix(m_matSpharaBabyMEGInnerFull, QString(QCoreApplication::applicationDirPath() + "/mne_x_plugins/resources/noisereduction/SPHARA/m_matSpharaBabyMEGInnerFull.txt"));
+        m_matSpharaMultFirst = Sphara::makeSpharaProjector(m_matSpharaBabyMEGInnerLoaded, indicesFirstBabyMEG, m_pFiffInfo->nchan, m_iNBaseFctsFirst, 0); //InnerLayer
     }
 
+    //Write final operator matrices to file
+    write_eigen_matrix(m_matSpharaMultFirst, QString(QCoreApplication::applicationDirPath() + "/mne_x_plugins/resources/noisereduction/SPHARA/m_matSpharaMultFirst.txt"));
```

Comment:
write_eigen_matrix should return a bool value true, when successful, false otherwise.